supervisor/resolution/checks/base.py

```diff
@@ -17,6 +18,13 @@ def __init__(self, coresys: CoreSys) -> None:
         """Initialize the checks class."""
         self.coresys = coresys
 
+    def __init_subclass__(cls, **kwargs):
+        """Initialize subclasses."""
+        super().__init_subclass__(**kwargs)
+        check = cls(None)
+        if check.slug not in CHECK_REGISTRY:
+            CHECK_REGISTRY.append(check)
+
```

Comment:
I think how Home Assistant does it and also perl/go kind of using it for dynamic loading is like https://github.com/home-assistant/supervisor/blob/main/supervisor/discovery/validate.py#L23 (Code is HA) and add an function like `def setup() -> CheckBase:` which would call to get it from. That way it need also less code

Reply:
neat!